src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -77,10 +77,39 @@ class FunSpec private constructor(builder: Builder) {
     }
 
     if (typeVariables.isNotEmpty()) {
-      codeWriter.emitTypeVariables(typeVariables)
-      codeWriter.emit(" ")
+      codeWriter.emitTypeVariables(typeVariables) {
```

Comment:
... this 8 line block would become a different 6 line block, with no repeated call to `emitSignature`:

```
if (typeVariables.isNotEmpty()) {
  codeWriter.emitTypeVariables(typeVariables);
  codeWriter.emit(" ");
}
emitSignature(codeWriter, enclosingName);
codeWriter.emitWhereBlock(typeVariables);
```

I think it’s more boring an easier to follow